File: nicktelsan/day01/slidingCountIncreases.js
import fs from 'fs'
import path from 'path'

const __dirname = path.resolve()

function main() {
  const input = fs.readFileSync(path.join(__dirname, 'day01/input'), 'utf8')
  const inputArray = input.split('\n')

  const maxLength = inputArray.length - 2

  let increases = 0
  let prev = null

  for (let i = 0; i < maxLength; i++) {
    const sum =
      parseInt(inputArray[i]) +
      parseInt(inputArray[i + 1]) +
      parseInt(inputArray[i + 2])

    if (prev && sum > prev) {
      increases++
    }

    prev = sum
  }

  console.log(increases)
}

main()

File: nicktelsan/day01/slidingCountIncreases.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"strconv"
	"strings"
)

func main() {

    content, err := ioutil.ReadFile("input")

     if err != nil {
          log.Fatal(err)
     }

    inputArray := strings.Split(string(content), "\n")

		var maxLength = len(inputArray) - 2
		var increases = 0

		for i := 1; i < maxLength; i++ {
			w, _ := strconv.Atoi(inputArray[i - 1])
			x, _ := strconv.Atoi(inputArray[i])
			y, _ := strconv.Atoi(inputArray[i + 1])
			z, _ := strconv.Atoi(inputArray[i + 2])

			current := x + y + z
			prev := w + x + y

			if( current > prev) {
				increases++
			}
		}
		fmt.Println(increases)
}